### PR TITLE
Update install guide with new numpy versions

### PR DESCRIPTION
**Pull request by emcastillo**

### CODE REVIEW

File: docs/source/install.rst [hunk 1]
```diff
@@ -23,11 +23,11 @@ Requirements
...
 ~~~~~~~~~~~~~~~~~~~
 
-NumPy/SciPy-compatible API in CuPy v10 is based on NumPy 1.20 and SciPy 1.6, and has been tested against the following versions:
+NumPy/SciPy-compatible API in CuPy v10 is based on NumPy 1.21 and SciPy 1.7, and has been tested against the following versions:
 
-* `NumPy <https://numpy.org/>`_: v1.17 / v1.18 / v1.19 / v1.20
+* `NumPy <https://numpy.org/>`_: v1.17 / v1.18 / v1.19 / v1.20 / v1.21
 
-* `SciPy <https://scipy.org/>`_ (*optional*): v1.4 / v1.5 / v1.6
+* `SciPy <https://scipy.org/>`_ (*optional*): v1.4 / v1.5 / v1.6 / v.1.7
```
Comment (kmaehashi) [inline docs/source/install.rst:30]:
```suggestion
* `SciPy <https://scipy.org/>`_ (*optional*): v1.4 / v1.5 / v1.6 / v1.7
```

File: docs/source/install.rst [hunk 2]
```diff
@@ -23,11 +23,11 @@ Requirements
 Python Dependencies
 ~~~~~~~~~~~~~~~~~~~
 
-NumPy/SciPy-compatible API in CuPy v10 is based on NumPy 1.20 and SciPy 1.6, and has been tested against the following versions:
+NumPy/SciPy-compatible API in CuPy v10 is based on NumPy 1.21 and SciPy 1.7, and has been tested against the following versions:
```
Comment (kmaehashi) [inline docs/source/install.rst:26]: This needs to be backported excluding this line.

**Review by kmaehashi — CHANGES_REQUESTED** [review]

File: docs/source/install.rst (hunk 2 above)
Comment (emcastillo) [reply docs/source/install.rst:26]: let's do it manually

**Review by toslunar — COMMENTED** [review]
How about updating also `docs/requirements.txt` (to update the comparison table, e.g. `scipy.special.ndtri_exp` from SciPy 1.7)?

**Comment by kmaehashi** [comment]
Added that to the task list https://github.com/cupy/cupy/wiki/Actions-Needed-for-Dependency-Update

> Update `docs/requirements.txt` to use the latest versions.

**Comment by emcastillo** [comment]
bumped docs requirements

**Review by asi1024 — APPROVED** [review]

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by kmaehashi** [comment]
Please ignore the backport bot error

**Pull request merged**